### PR TITLE
Fix typos in src subdirectory

### PR DESCRIPTION
**Pull request by luzpaz**
Found via `codespell -q 3 -S *.po,*.pot,*.svg,*.ts,./.git/logs,./share,./docs/man/es,./configs/attic,*_fr.*,*_es.*,README_es,./src/hal/classicladder -L ans,ba,breal,bu,bulle,busses,childs,componentes,currenty,doubleclick,dout,dum,etxt,extint,faktor,fo,fpr,halp,ihs,inout,inport,mot,parm,parms,propertys,ro,seh,ser,smoe,te,tey,trough,ue,ure,wille,wont,wonte`

### CODE REVIEW

File: src/hal/drivers/hal_gm.c [hunk 1]
```diff
@@ -1840,7 +1840,7 @@ RS485(void *arg, long period)
 	  {
 	    if((device-> RS485_mgr.ID[2*i]) != ((temp_u32 >> 8) & 0xff)) 
             {
-              //RS485 module falled off, error
+              //RS485 module failed off, error
```
Comment (hansu) [inline src/hal/drivers/hal_gm.c:1843]: probably "fell off" is the correct version

File: src/hal/drivers/hal_gm.c [hunk 2]
```diff
@@ -1854,7 +1854,7 @@ RS485(void *arg, long period)
 	  {
 	    if((device-> RS485_mgr.ID[2*i+1]) != ((temp_u32 & 0xff000000)>>24))
             {
-              //RS485 module falled off, error
+              //RS485 module failed off, error
```
Comment (hansu) [inline src/hal/drivers/hal_gm.c:1857]: probably "fell off" is the correct version

**Comment by luzpaz** [comment]
JFYI, there is another branch with misc. fixes at https://github.com/LinuxCNC/linuxcnc/compare/master...luzpaz:linuxcnc:typos-misc

Should I send a PR for that or further whittle it down?

File: src/hal/drivers/hal_gm.c (hunk 1 above)
Comment (hansu) [reply src/hal/drivers/hal_gm.c]: Or maybe just `//RS485 module failed, error` ?
"Fall off" does not make much sense in this context IMHO.

**Pull request merged**